File: torre-backend/src/main/kotlin/co/torre/goal/domain/Goal.kt
package co.torre.goal.domain

import co.torre.listitem.domain.ListItem
import java.util.*
import javax.persistence.*

@Entity
@Table(name="goals")
data class Goal (
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int,
        var peopleId: String,
        var interestId: String,
        var interestName: String,
        @OneToOne
        @JoinColumn(name = "level_id", referencedColumnName = "id")
        var level: ListItem?,
        var expirationDate: Date?,
        var score: Int?
)

File: torre-fe/src/app/pages/people/dto/goal.ts
import {ListItem} from '../../../core/dto/list-item';

export class Goal {
  constructor(
    public peopleId: string,
    public interestId: string,
    public interestName: string,
    public level: ListItem,
    public expirationDate: Date,
    public score: number,
    public id?: number,
  ) {
  }
}
